okpt/diff/diff.py

```diff
@@ -0,0 +1,141 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+
+#   http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Provides the Diff class."""
+
+from typing import Any, Dict, Tuple
+
+
+class InconsistentTestResultsError(Exception):
```

Comment:
Just wondering what the benefit of having these custom exceptions, given that each is only used once, they do not supply additional methods/logging, and they are not caught anywhere?

Reply:
I guess it would be fine to consolidate them as `InvalidTestResultsError`. My intention was to have one exception per validation check (type check, structure checks) to make it clear what the validation was doing and what went wrong. Should I just have one exception instead?

Reply:
Yes I think one validation exception with ability to pass in custom message is good.

Reply:
Done